src/js/easymde.js

```diff
@@ -1145,58 +1148,114 @@ var toolbarBuiltInButtons = {
     'bold': {
         name: 'bold',
         action: toggleBold,
-        className: 'fa fa-bold',
+        className: {
```

Comment:
Changing the className to this new object would introduce a **breaking change** for people that use the custom toolbar. 
I'd rather not break compatibility at this time to make the transition from SimpleMDE -> EasyMDE as painless as possbile.

We could introduce an `icon` field in the toolbar button options, that looks like this:
```js
// FontAwesome icon
'bold': {
    name: 'bold',
    action: toggleBold,
    icon: {
        fa: '<i class="fa fa-bold"></i>',
        material: '<i class="material-icons">format_bold</i>'
    },
    ...
},
```

This way we can simply pick the correct icon depending on the chosen (or default) iconsSet.
One thing to keep in mind though is to still allow overwriting the FontAwesome icon from `className` (which won't work with material icons).

We might be making the API unnecessarily complicated, perhaps we should delay the feature until a full rewrite is possible.

Reply:
What about this?

```js
'bold': {
    name: 'bold',
    action: toggleBold,
    icon: {
        'fa': {'className': 'fa fa-italic'},
        'material': {'className': 'material-icons', 'textContent': 'format_italic'},
    }
    ...
},
```

And something like this (not tested) in [`createIcon()`](https://github.com/Ionaru/easy-markdown-editor/pull/55/files#diff-9d17411f8015dc0eaff6773487fbb49bL118):

```js
var icon = document.createElement('i');
if(options.icon) {
    icon.className = options.icon[iconsSet || 'fa']['className'] || options.className;
    icon.textContent = options.icon[iconsSet || 'fa']['textContent'];
} else {
    icon.className = options.className
}